python/ray/serve/_private/common.py

```diff
@@ -33,6 +33,8 @@ class ApplicationStatus(str, Enum):
     DEPLOYING = "DEPLOYING"
     RUNNING = "RUNNING"
     DEPLOY_FAILED = "DEPLOY_FAILED"
+    DELETING = "DELETING"
+    NOT_EXISTED = "NOT_EXISTED"
```

Comment:
```suggestion
    INVALID = "INVALID"
```

We can also use `INVALID` for future states where the status requested is invalid for some reason. The user can find out why by looking at the status message. This lets us introduce new status behavior without updating this enum too frequently.

Reply:
I think this is an arguable point. I am afraid of using INVALID (a little too vague) and it will cause confusion on the dashboard side to handle different use cases in the future.

Reply:
Is this intended to be used in the dashboard for an application that doesn't exist? We shouldn't introduce a new state for that, we can just return a `404` response.